### PR TITLE
[FUSEQE-7781] Autodiscovery AMQ streams test

### PR DESCRIPTION
**Pull request by Stefan365**
//test: `@db-to-kafka-to-db`

### CODE REVIEW

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
Integration - DB to DB | from database to kafka topic 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mkralik3** [comment]
retest this please

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
Integration - DB to DB | from database to kafka topic 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by Stefan365** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

File: ui-tests/src/test/java/io/syndesis/qe/steps/other/KafkaSteps.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import cucumber.api.java.en.Given;
+import io.cucumber.datatable.DataTable;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class KafkaSteps {
+    private final NameConnectionSteps nameConnectionSteps = new NameConnectionSteps();
+    private final CommonSteps commonSteps = new CommonSteps();
+    private final Connections connectionsPage = new Connections();
+    @Autowired
```
[the remark this answers is not in the document]
Comment (Stefan365) [reply ui-tests/src/test/java/io/syndesis/qe/steps/other/KafkaSteps.java]: right. it is not.

File: ui-tests/src/test/resources/features/integrations/db-to-kafka.feature [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+
+#    B.kafka to db:
+    When navigate to the "Home" page
+    And click on the "Create Integration" link to create a new integration.
+    Then check visibility of visual integration editor
+    And check that position of connection to fill is "Start"
+
+    # select kafka connection as 'to' point
+    Then check visibility of page "Choose a Finish Connection"
```
[the remark this answers is not in the document]
Comment (Stefan365) [reply ui-tests/src/test/resources/features/integrations/db-to-kafka.feature]: yes, it should be Start Connection, however, mistake is on comment only.

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
Integration - Database | Smoke - Periodic invocation to Insert 
Integration - DB to DB | from database to kafka topic 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

File: ui-tests/src/test/java/io/syndesis/qe/steps/other/KafkaSteps.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Stefan365) [reply ui-tests/src/test/java/io/syndesis/qe/steps/other/KafkaSteps.java]: it is necessary due to spring context usage. Constructor is redundant.

**Comment by Stefan365** [comment]
retest this please

File: utilities/src/main/java/io/syndesis/qe/resource/impl/Kafka.java [hunk 3]
```diff
@@ -47,12 +57,67 @@ public boolean isReady() {
...
+            .endSubject()
+            .done();
+    }
+
+    private static void deleteClusterRole() {
+        final ClusterRoleBinding kafkaCrb = OpenShiftUtils.getInstance().rbac().clusterRoleBindings().list().getItems().stream()
+            .filter(crb -> crb.getMetadata().getName().equals(OCP_KAFKA_VIEW_ROLE)).findFirst().get();
+        OpenShiftUtils.getInstance().rbac().clusterRoleBindings().delete(kafkaCrb);
+        final ClusterRole kafkaCr = OpenShiftUtils.getInstance().rbac().clusterRoles().list().getItems().stream()
+            .filter(cr -> cr.getMetadata().getName().equals(OCP_KAFKA_VIEW_ROLE)).findFirst().get();
```
Comment (avano) [inline utilities/src/main/java/io/syndesis/qe/resource/impl/Kafka.java]:
```
final ClusterRoleBinding kafkaCrb = OpenShiftUtils.getInstance().rbac().clusterRoleBindings().withName(OCP_KAFKA_VIEW_ROLE).get()

and

final ClusterRole kafkaCr = OpenShiftUtils.getInstance().rbac().clusterRoles().withName(OCP_KAFKA_VIEW_ROLE).get()

```

File: utilities/src/main/java/io/syndesis/qe/utils/DbUtils.java [hunk 4]
```diff
@@ -118,6 +118,20 @@ public int getCountOfInvokedQuery(String query) {
         return records;
     }
 
+    public boolean containsExpected(String query, String expected) {
+        boolean contains = false;
```
Comment (avano) [inline utilities/src/main/java/io/syndesis/qe/utils/DbUtils.java]: remove this as it is not updated and `return false` in the end

File: utilities/src/main/java/io/syndesis/qe/bdd/validation/DbValidationSteps.java [hunk 5]
```diff
@@ -186,6 +186,12 @@ public void checkValuesExistInTable(String query) {
             "Query has no output.");
     }
 
+    @Then("^.*checks that query \"([^\"]*)\" has output \"([^\"]*)\"$")
```
Comment (avano) [inline utilities/src/main/java/io/syndesis/qe/bdd/validation/DbValidationSteps.java]: can't you use `@Then("^.*checks? that query \"([^\"]*)\" has \"(\\w+)\" output$")`?

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
Integration - DB to DB | from database to kafka topic 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

File: utilities/src/main/java/io/syndesis/qe/bdd/validation/DbValidationSteps.java (hunk 5 above)
Comment (Stefan365) [reply utilities/src/main/java/io/syndesis/qe/bdd/validation/DbValidationSteps.java]: ok

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
Integration - DB to DB | from database to kafka topic 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by Stefan365** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by Stefan365** [comment]
retest this please

**Comment by Stefan365** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by Stefan365** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Pull request merged**